### ISSUE TITLE
Sorted Usings Analyzer

### ISSUE THREAD

**Issue by golddove** [dotnet/roslyn#77146]
**Brief description:**

Similar code style rule already exists (dotnet_sort_system_directives_first), but there is no analyzer (see [#77024](https://github.com/dotnet/roslyn/issues/77024#issuecomment-2634891811)).

This is confusing as it is listed under [IDE0055](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0055) in MS docs, and looks just like other code style rules which do have analyzers (allowing build enforcement and IDE warning/error markings).

Ideally, also introduce a new code style rule (dotnet_sort_imports) which enables sorted usings independent of System usings order.

Any kind of analyzer that checks for sorted usings would be super useful for avoiding constant code changes from devs who haven't enabled Code Cleanup on save, and would make it easier to enforce it on cli builds too.

**Languages applicable:**

Both, or C#-only

**Code example that the analyzer should report:**

```
Microsoft.Namespace2
Microsoft.Namespace1
```

or

```
Microsoft.Namespace1
System.Namespace2
Microsoft.Namespace3
```

**Additional information:**

Any more additional information you would like to add.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
